### PR TITLE
AZ AOSM Extension Refactor

### PR DESCRIPTION
**Pull request by jordlay**
---
- History.RST and setup.py up to date for release of 2.0.0b
- Large scale refactor of az aosm extension
- Logic reviewed by responsible team
- Good unit test coverage
- Working integration tests using ScenarioTest

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by pjw711** [comment]
@jordlay , 

- See https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md . According to that the version we're releasing should be 2.0.0b1. It's a new preview version with breaking changes. 
-  The existing published version is still marked as preview - good :)

`{
    "experimental": false,
    "installed": true,
    "name": "aosm",
    "preview": true,
    "summary": "Microsoft Azure Command-Line Tools Aosm Extension",
    "version": "1.0.0b3"
  },`

File: src/aosm/azext_aosm/aaz/latest/aosm/publisher/artifact_manifest/_update_state.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "aosm publisher artifact-manifest update-state",
+    is_preview=True,
+)
+class UpdateState(AAZCommand):
+    """Update state for artifact manifest.
```
Comment (kairu-ms) [inline src/aosm/azext_aosm/aaz/latest/aosm/publisher/artifact_manifest/_update_state.py:19]: Please add examples for those aaz commands. You can reference this [doc](https://azure.github.io/aaz-dev-tools/pages/usage/workspace-editor/#command-examples)

Comment (Cyclam) [reply src/aosm/azext_aosm/aaz/latest/aosm/publisher/artifact_manifest/_update_state.py:19]: Thank you and done.

**Comment by Cyclam** [comment]
@kairu-ms , we've added examples as requested, could you review again, please?

**Comment by yonzhan** [comment]
Please fix CI issues

**Review by kairu-ms — APPROVED** [review]

**Comment by Cyclam** [comment]
Thanks @kairu-ms . Do we need to do anything else for this to merge, or is there another button you need to press?

**Comment by kairu-ms** [comment]
> Thanks @kairu-ms . Do we need to do anything else for this to merge, or is there another button you need to press?

PR merged

**Pull request merged**